internal/databases/postgres/tar_interpreter.go

```diff
@@ -118,6 +120,24 @@ func (tarInterpreter *FileTarInterpreter) Interpret(fileReader io.Reader, fileIn
 			return errors.Wrapf(err, "Interpret: failed to create hardlink %s", targetPath)
 		}
 	case tar.TypeSymlink:
+		// In some edge cases symlinks end up in the delta backups multiple times.
+		// os.Symlink cannot replace symlinks that already exist, and we want the latest symlink in our final restore.
+		// Therefore, we check symlinks in the tar file to see if the already exist and already are symlinks in the
+		// restore in which case we replace it (remove before creation).
+		if fi, err := os.Lstat(targetPath); os.IsNotExist(err) {
+			tracelog.DebugLogger.Printf("%s does not yet exist", targetPath)
+		} else if fi.Mode()&fs.ModeSymlink != 0 {
```

Comment:
I guess that this will cause the nil pointer dereference if `err` is not nil and `fi` is nil.

Reply:
Agreed, err!=nil should be handled too.
Logging using ErrorLogger and returning error now...